File: app/src/main/java/com/photoeditor/sketch/photo/maker/editor/sketchphotoeditor/pencilsketching/sketchphoto/ragnarok/SoftGlowFilter.java
package com.photoeditor.sketch.photo.maker.editor.sketchphotoeditor.pencilsketching.sketchphoto.ragnarok;

import android.graphics.Bitmap;

class SoftGlowFilter {
    static {
        System.loadLibrary("AndroidImageFilter");
    }

    static Bitmap softGlowFilter(Bitmap bitmap, double blurSigma) {
        int width = bitmap.getWidth();
        int height = bitmap.getHeight();

        int[] pixels = new int[width * height];
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height);

        int[] returnPixels = NativeFilterFunc.softGlow(pixels, width, height, blurSigma);

        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888);
    }
}

File: app/src/main/java/com/sketch/ragnarok/SoftGlowFilter.kt
package com.sketch.ragnarok

import android.graphics.Bitmap

internal object SoftGlowFilter {
    init {
        System.loadLibrary("AndroidImageFilter")
    }

    fun softGlowFilter(bitmap: Bitmap, blurSigma: Double): Bitmap {
        val width = bitmap.width
        val height = bitmap.height
        val pixels = IntArray(width * height)
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height)
        val returnPixels = NativeFilterFunc.softGlow(pixels, width, height, blurSigma)
        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888)
    }
}